app/src/main/java/tech/jpco/qen/viewModel/QenViewModel.kt

```diff
@@ -0,0 +1,128 @@
+    /* In theory, it keeps an "alive" data stream from UI interaction to UI output, allowing an event in flight to
+     * be resubscribed by a recreated view before it has finished its journey.
+     */
+    //TODO test the above description!
+    private val touchesIn = PublishSubject.create<DrawPoint>()
+    private val metaIn = PublishSubject.create<MetaEvent>()
+    private val mTouchesOut = PublishSubject.create<DrawPoint>()
+    private val mMetaOut = BehaviorSubject.create<SelectedPage>()
+
+    val touchesOut: Observable<DrawPoint> = mTouchesOut
+    val metaOut: Observable<SelectedPage> = mMetaOut
+
+
+    fun supply(repo: PagesRepository, touches: Observable<DrawPoint>, events: Observable<MetaEvent>) {
+        iLogger("supply")
+        if (!metaIn.hasObservers()) {
+            val stateStream = metaProcessor(repo, metaIn.doOnNext { iLogger("stateStream setup", it) }).share()
+                .apply { subscribe(mMetaOut) }
+            mMetaOut.onSubscribe(cd)
+
+            repo
+                .addTouchStream(
+                    touchesIn,
+                    stateStream
+                        .map { state -> state.current }
+                        .distinctUntilChanged()
+                )
+                .subscribe(mTouchesOut)
+            mTouchesOut.onSubscribe(cd)
+        }
+
+        events.doOnNext { iLogger("Received from UI", it) }.subscribe(metaIn)
+        touches.subscribe(touchesIn)
+    }
+
+    override fun onCleared() {
+        cd.dispose()
+        super.onCleared()
+    }
+
+    @VisibleForTesting
+    internal fun metaProcessor(
+        repo: PagesRepository,
+        inStream: Observable<MetaEvent>
+    ): Observable<SelectedPage> {
+        return repo
+            .getMaxPage(inStream.ofType(MetaEvent.CurrentPage::class.java).firstOrError().map { it.ar })
+            .doOnNext { iLogger("Max page is", it) }
+            .switchMap { maxPage ->
+                fun retrievePage(newCurrentPage: Int) =
+                    SelectedPage(
+                        newCurrentPage,
+                        maxPage,
+                        repo.getSelectedPagePoints(newCurrentPage),
+                        repo.getAR(newCurrentPage)
+                    )
+
+                fun Observable<MetaEvent.NewPage>.process() =
+                    subscribe {
+                        repo.addPage(it.ar)
+                    }
+
+                fun Observable<MetaEvent>.process() =
+                    scan(
+                        retrievePage(repo.mostRecentPage).also { iLogger("Default emitted", it) }
+                    ) { previousState: SelectedPage, incomingEvent: MetaEvent ->
+                        val currentPage = previousState.current
+
+                        when (incomingEvent) {
+                            is MetaEvent.CyclePage -> {
+                                retrievePage(
+                                    if (currentPage < maxPage)
```

Comment:
I'm not sure what the current intended behavior is, but currently there are 2 pages without ever clicking "New page." It would be an interesting exercise to enable/disable the "Cycle pages" button depending on if there are more than 1 page available. Perhaps the current UX is what you want though, but adding enabling and disabling of the cycle button until the user presses the new page button would be some added complexity to the "meta" behavior.